ui/components/app/gas-customization/advanced-gas-inputs/advanced-gas-inputs.component.js

```diff
@@ -52,22 +52,32 @@ export default class AdvancedGasInputs extends Component {
     }
   }
 
-  onChangeGasLimit = (e) => {
-    this.setState({ gasLimit: e.target.value });
-    this.changeGasLimit({ target: { value: e.target.value } });
+  onChangeGasLimit = ({ target: { value } }) => {
+    const newGasLimit = Number(value);
+    if (isNaN(newGasLimit) || newGasLimit < 0) {
```

Comment:
The `isNaN` check is redundant I think, since the fact that this is a `type=number` component guarantees the value gets coerced to a valid floating point number. I guess it doesn't hurt to check anyway though in case the type of the input is changed.